.vib/java/goss/java.yaml

```diff
@@ -0,0 +1,12 @@
+command:
+  # We need to parse the printed version differently depending on the version
+  # Java 1.8 does not support --version, with -version printing to stderr
+  # Major versions are often shown as "x" instead of "x.0.0"
+  check-app-version:
+    exec: java -version 2>&1 | sed "s/+/-/g" | sed "s/0_//g" | grep "$(echo $APP_VERSION | sed -E 's|([0-9]+)\.0\.0$|\1|g')"
```

Comment:
I would explain in a comment the reasoning behind the substitutions `s/+/-/g` and `s/0_//g` as they can be a little bit obscure (specially the last one)